docs/src/manuals/custom-node.md

```diff
@@ -0,0 +1,338 @@
+
+
+## Problem statement
+
+
+Jane wants to determine whether a coin is a fair coin, meaning that is equally likely to land on heads or tails. In order to determine this, she will throw the coin $K=20$ times and write down how often it lands on heads and tails. The result of this experiment is a realization of the underlying stochastic process. Jane models the outcome of the experiment $x_k\in\{0,1\}$ using the Bernoulli distribution as
+
+$$p(x_k \mid \pi) = \mathrm{Ber}(x_k \mid \pi) = \pi^{x_k} (1-\pi)^{1-x_k},$$
+
+where $\pi \in[0,1]$ denotes the probability that she throws heads, also known as the success probability. Jane also has a prior belief (initial guess) about the value of $\pi$ which she models using the Beta distribution as
+
+$$p(\pi) = \mathrm{Beta}(\pi \mid 4, 8).$$
+
+With this prior belief, the total probabilistic model that she has for this experiment is given by
+
+$$p(x_{1:K}, \pi) = p(\pi) \prod_{k=1}^K p(x_k \mid \pi).$$
+
+Jane is interested in determining the fairness of the coin. Therefore she aims to infer (calculate) the posterior belief of $\pi$, $p(\pi \mid x_{1:K})$, denoting how $\pi$ is distributed after we have seen the data.
+
+
+---
+
+
+## Step 1: Creating the custom node
+
+
+> Note: In this example we will assume that the `Bernoulli` node and distribution do not yet exist.
+
+
+First things first, let's import `RxInfer`:
+
+```julia
+using RxInfer
+```
+
+
+
+In order to define a custom node using the `@node` macro from `RxInfer`, we need the following three arguments:
+1. The name of the node. (`::Type`)
+2. Whether the node is `Deterministic` or `Stochastic`.
+3. The interfaces of the node and any potential aliases. (`::Vector`)
+
+
+For the name of the node we wish to use `MyBernoulli` in this tutorial (`Bernoulli` already exists). However, the corresponding distribution does not yet exist. Therefore we need to specify it first as
+
+```julia
+# struct for Bernoulli distribution with success probability π
+struct MyBernoulli{T <: Real} <: ContinuousUnivariateDistribution
+    π :: T
+    new(π :: Real) = 0 ≤ π ≤ 1 ? MyBernoulli(π) : throw(ArgumentError("π must be between 0 and 1"))	
+end
+
+# for simplicity, let's also specify the mean of the distribution
+Distributions.mean(d::MyBernoulli) = d.π
+```
+
+
+
+In this case the distribution also has an entry in the struct, however, this is not necessary as long as the name of the distribution is a `Type`. The custom node created with `struct NewNode end` would also work fine.
```

Comment:
Perhaps, we should mention that existing functions can be used as well, but with the `typeof(foo)` instead (because it is literally the name of their `Type`).